agent/src/main/scala/za/co/absa/atum/agent/model/Measure.scala

```diff
@@ -17,182 +17,156 @@
 import za.co.absa.atum.model.dto.MeasureResultDTO.ResultValueType
-import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
 
 /**
  *  Type of different measures to be applied to the columns.
  */
-sealed trait Measure extends MeasurementProcessor with MeasureType {
-  val controlCol: String
+sealed trait Measure {
+  val measureName: String
+  def controlColumns: Seq[String]
 }
 
-trait MeasureType {
-  val measureName: String
-  val resultValueType: ResultValueType.ResultValueType
+case class CustomMeasure(measureName: String, controlCols: Seq[String]) extends Measure {
+  override def controlColumns: Seq[String] = controlCols
 }
 
-object Measure {
+trait AtumMeasure extends Measure with MeasurementProcessor {
+  val resultValueType: ResultValueType.ResultValueType
+}
 
-  private val valueColumnName: String = "value"
+object AtumMeasure {
 
-  val supportedMeasures: Seq[MeasureType] = Seq(
-    RecordCount,
-    DistinctRecordCount,
-    SumOfValuesOfColumn,
-    AbsSumOfValuesOfColumn,
-    SumOfHashesOfColumn
+  val supportedMeasureNames: Seq[String] = Seq(
+    RecordCount.measureName,
+    DistinctRecordCount.measureName,
+    SumOfValuesOfColumn.measureName,
+    AbsSumOfValuesOfColumn.measureName,
+    SumOfHashesOfColumn.measureName
   )
-  val supportedMeasureNames: Seq[String] = supportedMeasures.map(_.measureName)
 
-  case class RecordCount private (
-    controlCol: String,
-    measureName: String,
-    resultValueType: ResultValueType.ResultValueType
-  ) extends Measure {
+  case class RecordCount private (measureName: String) extends AtumMeasure {
+    private val columnExpression = count("*")
 
     override def function: MeasurementFunction =
       (ds: DataFrame) => {
-        val resultValue = ds.select(col(controlCol)).count().toString
-        ResultOfMeasurement(resultValue, resultValueType)
+        val resultValue = ds.select(columnExpression).collect()
+        ResultOfMeasurement(resultValue(0).toString, resultValueType)
       }
-  }
-  object RecordCount extends MeasureType {
-    def apply(controlCol: String): RecordCount = RecordCount(controlCol, measureName, resultValueType)
 
-    override val measureName: String = "count"
+    override def controlColumns: Seq[String] = Seq.empty
     override val resultValueType: ResultValueType.ResultValueType = ResultValueType.Long
   }
+  object RecordCount {
+    private[agent] val measureName: String = "count"
+    def apply(): RecordCount = RecordCount(measureName)
+  }
+
+  case class DistinctRecordCount private (measureName: String, controlCols: Seq[String]) extends AtumMeasure {
+    require(controlCols.nonEmpty, "At least one control column has to be defined.")
 
-  case class DistinctRecordCount private (
-    controlCol: String,
-    measureName: String,
-    resultValueType: ResultValueType.ResultValueType
-  ) extends Measure {
+    private val columnExpression = countDistinct(col(controlCols.head), controlCols.tail.map(col): _*)
```

Comment:
I'm still not sure if this is what we want. I thought that it's about 'distinct -> count' operation, performed on a single column. Do you remember what was used in the old Atum & what can our users require @benedeki ?

Reply:
With the suggested code it can be performed on single column or multiple columns. If this particular measure is expected to work differently please let me know. @benedeki

Reply:
I like the universality of supporting multi-column uniqueness 👍